### ISSUE TITLE
Run ERROR

### ISSUE THREAD

**Issue by GitRealNB** [KumapowerLIU/CSA-inpainting#28]
pic = (torch.cat([real_A, real_B,fake_B], dim=0) + 1) / 2.0
            torchvision.utils.save_image(pic_1, '%s/Epoch_(%d)_(%dof%d).jpg' % (
            save_dir, epoch, total_steps + 1, len(dataset_train)), nrow=2)

pic_1  not defined, 是不是要把pic_1改成pic？
按如上修改后，save_dir not defined 这是为啥呢？是代码执行的先后顺序有问题吗？

**Comment by TS-Logic** [comment]
fix pic_1 with pic.

**Comment by TS-Logic** [comment]
type an absolute path instead of the save_dir variable.

**Comment by GitRealNB** [comment]
> fix pic_1 with pic.

想请教一下  怎么设置epoch个数啊？目前是120？

**Comment by GitRealNB** [comment]
> type an absolute path instead of the save_dir variable.
wana ask where to change epoch？ now sets as 120

**Comment by TS-Logic** [comment]
Check out the Opion class. To change the default epoch, change the value of opt.niter, opt.niter_decay.

**Comment by GitRealNB** [comment]
> Check out the Opion class. To change the default epoch, change the value of opt.niter, opt.niter_decay.

thanks! I am sorry for bother you again! I don't know why the mask has been stored in Maskroot, why always masktype = 'center'?
The mask in the folder does not seem to be used

**Comment by GitRealNB** [comment]
> > Check out the Opion class. To change the default epoch, change the value of opt.niter, opt.niter_decay.
> 
> thanks! I am sorry for bother you again! I don't know why the mask has been stored in Maskroot, why always masktype = 'center'?
> The mask in the folder does not seem to be used

why the testdate only output just one figure?

**Comment by TS-Logic** [comment]
If you change the value of self.mask_type to 'random', mask in the folder is used to train the model.

**Comment by 18130282610** [comment]
> pic = (torch.cat([real_A, real_B,fake_B], dim=0) + 1) / 2.0 torchvision.utils.save_image(pic_1, '%s/Epoch_(%d)_(%dof%d).jpg' % ( save_dir, epoch, total_steps + 1, len(dataset_train)), nrow=2)
> 
> pic_1 not defined, 是不是要把pic_1改成pic？ 按如上修改后，save_dir not defined 这是为啥呢？是代码执行的先后顺序有问题吗？

你好！请问你这个 save_dir 未定义的问题是怎么解决的？

**Comment by 18130282610** [comment]
> 
Hi! Do you mean this? 
torchvision.utils.save_image(pic, '%s/Epoch_(%d)_(%dof%d).jpg' % (
            r'F:\save', epoch, total_steps + 1, len(dataset_train)), nrow=2)

**Comment by TT-mouse** [comment]
> > 
> 
> Hi! Do you mean this? torchvision.utils.save_image(pic, '%s/Epoch_(%d)_(%dof%d).jpg' % ( r'F:\save', epoch, total_steps + 1, len(dataset_train)), nrow=2)



> > pic = (torch.cat([real_A, real_B,fake_B], dim=0) + 1) / 2.0 torchvision.utils.save_image(pic_1, '%s/Epoch_(%d)_(%dof%d).jpg' % ( save_dir, epoch, total_steps + 1, len(dataset_train)), nrow=2)
> > pic_1 not defined, 是不是要把pic_1改成pic？ 按如上修改后，save_dir not defined 这是为啥呢？是代码执行的先后顺序有问题吗？
> 
> 你好！请问你这个 save_dir 未定义的问题是怎么解决的？

请问save_dir 未定义这个问题您是怎么解决的？
